### ISSUE TITLE
Stop cAdvisor from monitoring docker containers

### ISSUE THREAD

**Issue by ncapps** [google/cadvisor#2848]
**Summary**
We have both Kubernetes and Docker running in the same environment. cAdvisor is running as part of the kubelet binary. Kubernetes is using CRI-O as our CRI. Both `/var/lib/docker` and `/var/lib/containers` are backed by a ZFS file system.

cAdvisor is gathering metrics for the Docker CRI resulting in the execution `zfs list` on an interval. This operation is resource intensive. We do not need cAdvisor to gather metrics for the Docker CRI.

I would like to be able to disable gathering metrics for the Docker CRI.

**Comment by iwankgb** [comment]
@ncapps let me rephrase your problem to make sure that I understand what you want to achieve: you are using kubelet `/metrics` endpoint to collect information about containers and you do not want this endpoint to take non-kubernetes containers into consideration?

**Comment by ncapps** [comment]
@iwankgb Yes, that's correct

**Comment by iwankgb** [comment]
AFAIK recommended method is to use standalone cAdvisor rather than rely on endpoint that kubelet exposes. It would give you much more flexibility over cAdvisor configuration.

**Comment by filippog** [comment]
I'm in the same scenario, namely I'd like to have `cadvisor` collect metrics from the root cgroup and not talk to docker at all. We are rolling out `cadvisor` to a fleet of hosts where only a subset will be running `kubelet`, and at that stage I'm happy if `kubelet` exports metrics from docker containers.

**Comment by dims** [comment]
Closing this issue as it has been open since before January 1, 2024. If this is still relevant, please feel free to reopen or create a new issue.
